src/widget/wnumberrate.cpp

```diff
@@ -28,16 +28,19 @@ WNumberRate::WNumberRate(const char * group, QWidget * parent)
     setValue(0);
 }
 
-void WNumberRate::setValue(double /*dValue*/) {
-    double vsign = m_pRateControl->get() *
-            m_pRateRangeControl->get() *
-            m_pRateDirControl->get();
+void WNumberRate::setValue(double dValue) {
+    double digitFactor = pow(10, m_iNoDigits);
+    // Calculate percentage rounded to the number of digits specified by iNoDigits
+    double percentage = round((dValue - 1) * 100.0 * digitFactor) / digitFactor;
 
-    char sign = '+';
-    if (vsign < -0.00000001) {
+    QString sign(' ');
+    if (percentage > 0) {
+        sign = '+';
+    }
+    if (percentage < 0) {
         sign = '-';
     }
```

Comment:
Can you make this a function? Unless we need it elsewhere, this can go into the Anonymous namespace of `wnumberrate.cpp`.

Reply:
Do you mean like `getSignChar(): char` ?

Reply:
> sign should be implemented as a function `inline QChar sign(double value)` in an anonymous namespace using if/else if/else for the 3 different return paths

Reply:
ping

Reply:
```
inline QChar sign(double value) {
    sign = '';
    if (value > 0) {
        sign = '+';
    }
    if (value < 0) {
        sign = '-';
    }
    return sign;
}
```

Something like this?

I made this:

![image](https://user-images.githubusercontent.com/7236157/93024102-fe44c800-f5f3-11ea-92ca-f8dcc737881d.png)

Reply:
Something like this at the top of the cpp file (below includes/forward declarations)

```cpp
namespace {

inline QChar sign(double value) {
    if (value < 0) {
        return '-';
    }
    if (value > 0) {
        return '+';
    }
    return '';
}

} // anonymous namespace
```

Reply:
ok. Does `char sign = sign(percentage);` make sense?

Reply:
Why not use it directly in `setText()`? If you want to store the return value first, use `QChar`, not `char`.

Reply:
The calculation doesn't take the rate range and the rate direction into account. I'll just open a new PR to finally get this merged.